website/docs/guides/migrating-from-v2-to-v3.mdx

```diff
@@ -0,0 +1,87 @@
++ import { parse, validate, execute, subscribe } from 'graphql';
+
+- const getEnveloped = envelop([ ... ])
++ const getEnveloped = envelop({ parse, validate, execute, subscribe, plugins: [ ... ] })
+```
+
+### 2. Remove `useAsyncSchema` plugin
+
+This was a mistake from beginning as we cannot asynchronously `validate` and `parse` since with [`graphql`](https://github.com/graphql/graphql-js) these functions are synchronous in nature.
+
+### 3. Rename `useLazyLoadedSchema` to `useSchemaByContext`
+
+Oringinal name was very misleading since lazy loading could mean it can be asynchronous in nature. This plugin was renamed to better reflect its purpose. It is now called `useSchemaByContext`
+
+### 4. Remove `enableIf` utility
+
+This utility was used to enable plugins conditionally. For a better developer experience we have dropped this utility and favor more type safe way to conditionally enable plugins.
+
+```diff
+- import { envelop, useMaskedErrors, enableIf } from '@envelop/core'
++ import { envelop, useMaskedErrors } from '@envelop/core'
+import { parse, validate, execute, subscribe } from 'graphql'
+
+const isProd = process.env.NODE_ENV === 'production'
+
+const getEnveloped = envelop({
+	parse,
+   validate,
+   execute,
+   subscribe,
+   plugins: [
+     // This plugin is enabled only in production
+-    enableIf(isProd, useMaskedErrors())
++    isProd && useMaskedErrors()
+   ]
+ })
+```
+
+### 5. Update options for `useMaskedErrors` plugin
+
+- We _removed_ `handleValidationErrors` and `handleParseErrors` options since ONLY masking validation errors OR ONLY disabling introspection errors does not make sense, as both can be abused for reverse-engineering the GraphQL schema (see https://github.com/nikitastupin/clairvoyance for reverse-engineering the schema based on validation error suggestions). Instead you should use `useErrorHandler` plugin where you can access each phase and decide what to do with the error.
+- Renamed `formatError` to `maskError`
+
+### 6. Drop `useTiming` plugin
+
+### 7. Drop support for Node.js v12
+
+Node.js v12 is no longer supported by the Node.js team. https://github.com/nodejs/Release/#end-of-life-releases
+
+### 8. Remove `onResolverCalled`
+
+We were wrapping the `GraphQLSchema` object but with the new version we no longer want to depend on a specific implmentation. So we decided to drop this and instead provide a new plugin that will let you hook into this phase.
+
+```diff
+import { envelop, Plugin } from '@envelop/core';
++ import { useOnResolve } from '@envelop/on-resolve';
+
+import { onResolverCalled } from './my-resolver';
+
+function useResolve(): Plugin {
+  return {
+-   onResolverCalled: onResolverCalled,
++   onPluginInit: ({ addPlugin }) => {
++     addPlugin(useOnResolve(onResolverCalled));
++   },
+  };
+}
+
+const getEnveloped = envelop({
+  plugins: [
+    // ... other plugins ...
+    useResolve(),
+  ],
+});
+```
+
+PR: https://github.com/n1ru4l/envelop/pull/1500
```

Comment:
Make move this up top so the first line is [new major version](https://github.com/n1ru4l/envelop/pull/1500)